### DISCUSSION TITLE
Does Immich transcode HDR video to SDR?

### DISCUSSION THREAD

**Discussion by Zk2u** [immich-app/immich#5755] [Q&A] [answered]
Hi there! Thanks for this awesome project. I have a question: I'm using hardware transcoding setup using HEVC on a RTX 3070 Ti. If tone mapping is on, does it convert all my HDR videos to SDR? I'm a little confused to how transcoding actually works in Immich/its purpose.

**Comment by mertalev** [comment]
Transcoded videos are converted from HDR to SDR either way for browser and device compatibility. The tone-mapping setting just preserves colors better when doing this conversion.

**Reply by Zk2u** [reply]
Should I disable transcoding all together then to preserve HDR? Thanks for quick reply

**Reply by alextran1502** [reply]
If you disable transcoding, videos that are not mp4 cannot be played on the web

**Reply by Zk2u** [reply]
Is transcoding *only* for streaming to the browser then? It doesn't replace the originals? I want to keep everything in original quality, including HDR.

**Reply by mertalev** [reply]
It depends. If you mainly view videos through Chrome and not through Firefox or the Immich app, then disabling transcoding would be fine.

Firefox won't display HDR videos at all, and the colors will be wrong on mobile. (For the latter, we're working to switch to a different video player that can handle HDR videos better, but this is the current state of things.)

**Reply by zackpollard** [reply, marked as answer]
We always keep the originals of everything untouched. Transcoding and other generated content are all for the purposes of better displaying in the apps and web.

**Reply by Zk2u** [reply]
~~Thank you. I'm just trying to work out if Immich takes:

1. a Nextcloud Memories style approach of storing the videos as originals and then transcoding on demand to browsers. HDR originals are what is stored on the server. 
2. Transcodes my videos and replaces the originals in storage, then streams these to clients. HDR originals no longer exist on Immich

I'm all for enabling transcoding for (1) but will disable it if (2) is what's happening~~

**edit:** it seems (1) is what is occurring. Thank you!

**Reply by mertalev** [reply]
Oh, if that's what you're wondering then rest assured we never touch the originals. Transcoded videos are different files that live in the `encoded-videos` folder (as opposed to the `library` folder where the originals live).
